experimental/JuLie/schur_polynomials.jl

```diff
@@ -58,32 +52,21 @@ x_1^{λ_n} & x_2^{λ_n} & … & x_n^{λ_n}
 \end{vmatrix}
 ```
 """
-function schur_polynomial(lambda::Partition{T}, n::Int=sum(lambda);
-    ring::AbstractAlgebra.Ring=(n<=0 ? ZZ : begin # the ring for the return value
-                                  x = [string("x",string(i)) for i=1:n]
-                                  PolynomialRing(ZZ, x)[1]
-                                end
-                               )
-  ) where T<:Integer
+function schur_polynomial(lambda::Partition{T}, n::Int=sum(lambda)) where T<:Integer
```

Comment:
@ulthiel actually, why is the default value for `n` here `sum(lambda)`? I would have expected `length(lambda)`... 

In the definition of Schur polynomial I am familiar with, then given a partition of length $n$ I'd expect the Schur polynomial to be in $n$ indeterminates. So it is not even clear to me what happens when `n > length(lambda)`. Perhaps there is a more general definition which allows this?

What am I missing?

Reply:
See my comment above. True, the default should be length(lambda) and this is wrong several times.